tests/Dfc.CourseDirectory.WebV2.Tests/FeatureTests/TLevels/DeleteTLevelTests.cs

```diff
@@ -0,0 +1,487 @@
+
+            // Assert
+            response.EnsureSuccessStatusCode();
+
+            var doc = await response.GetDocument();
+            doc.GetElementByTestId("ViewEditLink").Should().BeNull();
+        }
+
+        [Fact]
+        public async Task GetDeleted_HasOtherLiveCourseRuns_DoesRenderViewEditCopyLink()
+        {
+            // Arrange
+            var tLevelDefinitions = await TestData.CreateInitialTLevelDefinitions();
+
+            var providerId = await TestData.CreateProvider(
+                ukprn: 12345,
+                providerType: ProviderType.TLevels,
+                tLevelDefinitionIds: tLevelDefinitions.Select(tld => tld.TLevelDefinitionId).ToArray());
+
+            var venueId = await TestData.CreateVenue(providerId, venueName: "T Level venue");
+
+            var tLevelDefinition = tLevelDefinitions.First();
+
+            var yourReference = "YOUR-REF";
+            var startDate = new DateTime(2021, 10, 1);
+
+            var tLevel = await TestData.CreateTLevel(
+                providerId,
+                tLevelDefinition.TLevelDefinitionId,
+                locationVenueIds: new[] { venueId },
+                yourReference: yourReference,
+                startDate: startDate,
+                createdBy: User.ToUserInfo());
+
+            await WithSqlQueryDispatcher(
+                dispatcher => dispatcher.ExecuteQuery(new DeleteTLevel() { TLevelId = tLevel.TLevelId }));
+
+            var anotherTLevel = await TestData.CreateTLevel(
+                providerId,
+                tLevelDefinitions.Last().TLevelDefinitionId,
+                locationVenueIds: new[] { venueId },
+                createdBy: User.ToUserInfo());
+
+            var request = new HttpRequestMessage(
+                HttpMethod.Get,
+                $"/t-levels/{tLevel.TLevelId}/delete/success?providerId={providerId}");
+
+            await User.AsProviderUser(providerId, ProviderType.FE);
+
+            var journeyInstance = CreateJourneyInstance(
+                tLevel.TLevelId,
+                new JourneyModel()
+                {
+                    TLevelName = tLevelDefinition.Name,
+                    ProviderId = providerId
+                });
+
+            journeyInstance.Complete();
+
+            // Act
+            var response = await HttpClient.SendAsync(request);
+
+            // Assert
+            response.EnsureSuccessStatusCode();
+
+            var doc = await response.GetDocument();
+            doc.GetElementByTestId("ViewEditLink").Should().NotBeNull();
+        }
+
+        private JourneyInstance<JourneyModel> CreateJourneyInstance(
+            Guid tLevelId,
+            JourneyModel journeyModel = null)
+        {
+            return CreateJourneyInstance(
+                "DeleteTLevel",
+                configureKeys: keysBuilder => keysBuilder
+                    .With("tLevelId", tLevelId),
+                journeyModel ?? new JourneyModel());
+        }
+    }
+}
```

Comment:
This is quite a big test file, is there any opportunity for good old fashioned DRY to make the intent clearer with less lines of code?